### ISSUE TITLE
Incorrect CustomStringConvertible implementations

### ISSUE THREAD

**Issue by lorentey** [apple/swift-collections#301]
`OrderedDictionary.description` does not use the same output style as `Dictionary` -- it prints its keys and values by calling `String(describing:)`, rather than the expected `String(reflecting:)`. This allows the contents of keys and values to confusingly intermix with the syntax of the description's structural output.

### Information

- **Package version:** release/1.0, release/1.1, main
- **Platform version:** n.a.
- **Swift version:** 5.9

### Checklist

- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] I've searched for [existing GitHub issues](https://github.com/apple/swift-collections/issues).

### Steps to Reproduce

```swift
    let d: OrderedDictionary = [
      "1: 2, 3": 4,
      "5: 6, 7": 8,
    ]
    print(d)
```

### Expected behavior

I expect the code to print output that distinguishes keys from values.

### Actual behavior

The sample code prints the following output:

```
[1: 2, 3: 4, 5: 6, 7: 8]
```

This is extremely confusing -- the values of the keys are allowed to interfere with the syntactic punctuation in the dictionary's structured description.

In contrast, the standard `Dictionary.description` uses `debugDescription` to print its contents, preventing such confusion:

```swift
    let d: Dictionary = [
      "1: 2, 3": 4,
      "5: 6, 7": 8,
    ]
    print(d) // ⟹ ["1: 2, 3": 4, "5: 6, 7": 8]
```

The output is not exactly crystal clear, but it is at least possible to e.g. count the number of elements in the dictionary.

**Comment by lorentey** [comment]
The descriptions are generated by code that is shared across all container types in this package (`_dictionaryDescription` and `_arrayDescription`), so they are all affected by this problem.

**Comment by lorentey** [comment]
I think it'd be a good idea to also drop the type names from `debugDescription` as part of resolving this -- generally it is a bad idea to do that as type information is readily available elsewhere (from context, for example).

Given the use of `debugDesctiption` in `Array`/`Set`/`Dictionary`'s `description` implementation, best practice is to keep `description` and `debugDescription` largely the same, except for escaping things that could potentially interfere with surrounding syntactic punctuation -- precisely like `String` does.

**Comment by lorentey** [comment]
Resolved in #302. `OrderedDictionary` (and all other collections in the package) now follow the practices of the stdlib.
